interface/cache_cow.py

```diff
@@ -0,0 +1,47 @@
+import time
+import numpy as np
+import os
+from astropy.time import Time
+import json
+import glob
+f = open(os.environ['DSAPOLDIR'] + "directories.json","r")
+dirs = json.load(f)
+f.close()
+
+logfile = dirs['logs'] + "cache_cow_log.txt"
+
+
+#run continuously, look for logs and caches that are more than a week old and delete them. At some point, want to make this backup to dsastorage?
+
+def main():
+    while True:
+        f = open(logfile,"w")
+        #list caches and log directories
+        allcaches = glob.glob(dirs['cache']+"*")
+        alllogs = glob.glob(dirs['logs']+"*T*")
+
+        #check which ones older than 1 week
+        lim = 7 #days
+        now = Time.now()
+        print("Scanning Cache...",file=f)
+        for cache in allcaches:
+            print(cache,file=f)
+            cachetime = Time(cache[len(dirs['cache']):len(dirs['cache'])+23],format='isot')
+            if np.abs((now-cachetime).value) >= lim:
+                os.system("rm -r " + cache)
+                print(now.isot + " cache_cow deleted CACHE " + cache[len(dirs['cache']):],file=f)
+
+        print("Scanning Logs...",file=f)
+        for log in alllogs:
+            print(log,file=f)
+            logtime = Time(log[len(dirs['logs']):],format='isot')
+            if np.abs((now-logtime).value) >= lim:
+                os.system("rm -r " + log)
```

Comment:
Does this need to be recursive?

Reply:
@caseyjlaw Yes, I had revised the logs to be written in time-stamped directories (theres multiple logfiles in each directory corresponding to different stages of the pipeline. Thanks for double checking though, I'll document here for reference: below is the layout for an example log dir:

```
2024-08-21T22:56:30.380/RMcal_logfile.txt     2024-08-21T22:56:30.380/dedisp_logfile.txt  2024-08-21T22:56:30.380/polcal_logfile.txt
2024-08-21T22:56:30.380/archive_logfile.txt   2024-08-21T22:56:30.380/dsapol_logfile.txt  2024-08-21T22:56:30.380/scatscint_logfile.txt
2024-08-21T22:56:30.380/beamform_logfile.txt  2024-08-21T22:56:30.380/filter_logfile.txt

'2024-08-21T22:56:30.380/RM_files':

'2024-08-21T22:56:30.380/scat_files':
```

The `RM_files` and `scat_files` subdirectories usually hold intermediate files for background processing (below), and otherwise are empty:

```
'RM_files/proc_1D_2024-08-09T13:43:37.740':
SNRs.npy  result.npy  trialRM.npy

'RM_files/proc_1D_2024-08-09T13:50:03.254':
SNRs.npy  result.npy  trialRM.npy

'RM_files/proc_2D_2024-08-09T13:50:06.976':
SNRs.npy  SNRs_full.npy  result.npy  trialRM.npy
```

```
'scat_files/proc_scat_2024-08-09T13:26:57.305':
240122aaag_mahi_MCMC_BIC.npy  240122aaag_mahi_MCMC_results.npy  240122aaag_mahi_MCMC_samples.npy

'scat_files/proc_scat_2024-08-09T13:40:18.482':
240122aaag_mahi_MCMC_BIC.npy  240122aaag_mahi_MCMC_results.npy  240122aaag_mahi_MCMC_samples.npy
```